### PR TITLE
Ianhelle/writing data provs doc 2022 03 14

### PR DESCRIPTION
**Pull request by ianhelle**

### CODE REVIEW

File: docs/source/data_acquisition/WritingADataProvider.rst [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+To implement a data provider you need to do the following:
+
+1. Write the driver
+2. Customize the driver (optional)
+3. Register the driver
+4. Add queries
+5. Create documentation
+6. Create unit tests
+
+7. Write the driver class
```
Comment (petebryan) [inline docs/source/data_acquisition/WritingADataProvider.rst:39]: Is this meant to be a 1?

**Review by petebryan — DISMISSED** [review]

File: docs/source/data_acquisition/WritingADataProvider.rst (hunk 1 above)
Comment (ianhelle) [reply docs/source/data_acquisition/WritingADataProvider.rst:39]: Fixed

File: docs/source/data_acquisition/WritingADataProvider.rst [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+This method is called from ``QueryProvider.connect`` and is used to authenticate to
+the data service. It takes and optional ``connect_str`` parameter and a ``kwargs``
+keyword argument dictionary.
+
+Any per-connection configuration settings can be read in here using the
+``DriverBase._get_config_settings(ProviderName)`` method. This returns the ``args``
+section of your configuration settings from ``msticpyconfig.yaml``.
+
+Some existing drivers use an API key to authenticate, some use name/password and others
+use Azure AAD. See :py:class:`KqlDriver <msticpy.data.drivers.kql_driver.KqlDriver>`
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply docs/source/data_acquisition/WritingADataProvider.rst:85]: Fixed

File: docs/source/data_acquisition/WritingADataProvider.rst [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+        DataEnvironment.Cybereason: ("cybereason_driver", "CybereasonDriver"),
+        DataEnvironment.Elastic: ("elastic_driver", "ElasticDriver"),
+        DataEnvironment.YourProvider: ("your_module", "YourDriverClassName"),
+    }
+
+
+4. Add queries
+--------------
+
+Create a folder in msticpy/data/queries with the name of your provider and
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply docs/source/data_acquisition/WritingADataProvider.rst:328]: Fixed

**Pull request merged**